### PR TITLE
EdgeFS: support for S3 buckets as DNS subdomain style

### LINKED ISSUE #2510: EdgeFS S3 bucket as DNS subdomain
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**

EdgeFS S3 protocol enhancement to support bucket as DNS subdomain implemented in separate Docker image. Need a way to select image via existing S3 CRD and also additional properties.

**What is use case behind this feature:**

**Environment**:

### PR DESCRIPTION
**Pull request by dyusupov**
**Description of your changes:**

Provided a way to configure s3type.

**Which issue is resolved by this Pull Request:**
Resolves #2510, #2681 

**Checklist:**
- [ ] Documentation has been updated, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](../blob/master/CONTRIBUTING.md#comments)

### CODE REVIEW

File: Documentation/edgefs-s3-crd.md [hunk 1]
```diff
@@ -117,7 +119,14 @@ efscli service create s3 s3Pepsi
 efscli service serve s3Pepsi Hawaii/Pepsi/bk1
 </pre>
 
-5. Create S3X CRDs
+In case of s3type set to `s3`, do not forget to configure default domain name:
+
+<pre>
```
Comment (galexrt) [inline Documentation/edgefs-s3-crd.md]: Please use ``` here.

File: cluster/examples/kubernetes/edgefs/cluster.yaml [hunk 2]
```diff
@@ -55,7 +55,7 @@ subjects:
   name: rook-edgefs-cluster
   namespace: rook-edgefs
 ---  
-apiVersion: apps/v1
+apiVersion: policy/v1beta1
```
Comment (galexrt) [inline cluster/examples/kubernetes/edgefs/cluster.yaml]: This fixes #2681, right? If this needs backport to v0.9, then the change needs to be in a different commit, as the other changes are introducing a new feature.

Comment (dyusupov) [reply cluster/examples/kubernetes/edgefs/cluster.yaml]: no need to backport

File: Documentation/edgefs-s3-crd.md (hunk 1 above)
Comment (dyusupov) [reply Documentation/edgefs-s3-crd.md]: done throughout!
